gcloud-java-examples/src/main/java/com/google/cloud/examples/datastore/DatastoreExample.java

```diff
@@ -56,15 +64,24 @@
   private static final String DEFAULT_ACTION = "display";
   private static final Map<String, DatastoreAction> ACTIONS = new HashMap<>();
 
-  private interface DatastoreAction {
-    void run(Transaction tx, Key userKey, String... args);
+  private abstract static class DatastoreAction<T> {
+
+    abstract void run(Transaction tx, Key userKey, T request) throws Exception;
 
-    String getRequiredParams();
+    abstract T parse(String... args) throws Exception;
+
+    protected String params() {
+      return "";
```

Comment:
This returns an empty string so that extending actions that take no arguments can avoid redefining.